dlt/destinations/impl/databricks/databricks.py

```diff
@@ -0,0 +1,336 @@
+        self.config: DatabricksClientConfiguration = config
+        self.sql_client: DatabricksSqlClient = sql_client
+        self.type_mapper = DatabricksTypeMapper(self.capabilities)
+
+    def start_file_load(self, table: TTableSchema, file_path: str, load_id: str) -> LoadJob:
+        job = super().start_file_load(table, file_path, load_id)
+
+        if not job:
+            job = DatabricksLoadJob(
+                file_path,
+                table["name"],
+                load_id,
+                self.sql_client,
+                stage_name=self.config.stage_name,
+                staging_credentials=(
+                    self.config.staging_config.credentials if self.config.staging_config else None
+                ),
+            )
+        return job
+
+    def restore_file_load(self, file_path: str) -> LoadJob:
+        return EmptyLoadJob.from_file_path(file_path, "completed")
+
+    def _create_merge_job(self, table_chain: Sequence[TTableSchema]) -> NewLoadJob:
+        return DatabricksMergeJob.from_table_chain(table_chain, self.sql_client)
+
+    def _create_merge_followup_jobs(self, table_chain: Sequence[TTableSchema]) -> List[NewLoadJob]:
+        return [DatabricksMergeJob.from_table_chain(table_chain, self.sql_client)]
+
+    # def _create_staging_copy_job(self, table_chain: Sequence[TTableSchema]) -> NewLoadJob:
+    #     return DatabricksStagingCopyJob.from_table_chain(table_chain, self.sql_client)
+
+    def _make_add_column_sql(
+        self, new_columns: Sequence[TColumnSchema], table_format: TTableFormat = None
+    ) -> List[str]:
+        # Override because databricks requires multiple columns in a single ADD COLUMN clause
+        return ["ADD COLUMN\n" + ",\n".join(self._get_column_def_sql(c) for c in new_columns)]
+
+    # def _create_optimized_replace_job(self, table_chain: Sequence[TTableSchema]) -> NewLoadJob:
+    #     return DatabricksStagingCopyJob.from_table_chain(table_chain, self.sql_client)
+
+    def _create_replace_followup_jobs(
+        self, table_chain: Sequence[TTableSchema]
+    ) -> List[NewLoadJob]:
+        if self.config.replace_strategy == "staging-optimized":
+            return [DatabricksStagingCopyJob.from_table_chain(table_chain, self.sql_client)]
+        return super()._create_replace_followup_jobs(table_chain)
+
+    def _get_table_update_sql(
+        self,
+        table_name: str,
+        new_columns: Sequence[TColumnSchema],
+        generate_alter: bool,
+        separate_alters: bool = False,
+    ) -> List[str]:
+        sql = super()._get_table_update_sql(table_name, new_columns, generate_alter)
+
+        cluster_list = [
+            self.capabilities.escape_identifier(c["name"]) for c in new_columns if c.get("cluster")
+        ]
+
+        if cluster_list:
+            sql[0] = sql[0] + "\nCLUSTER BY (" + ",".join(cluster_list) + ")"
+
+        return sql
+
+    def _execute_schema_update_sql(self, only_tables: Iterable[str]) -> TSchemaTables:
```

Comment:
why do you need to override this method? won't the one in the baseclass work?

Reply:
Now it does. Updated the baseclass to use `execute_many`. The base sqlclient implements it as:
https://github.com/dlt-hub/dlt/blob/sthor%2Fdatabricks/dlt/destinations/sql_client.py#L122-L139  depending on capabilities